run_test

```diff
@@ -18,10 +18,31 @@ tmp_file=$(mktemp)
 
 # Invoke mimpirun with appropriate timeout and arguments
 # "|| true" used to ignore ./mimpirun return code
-timeout "$TIMEOUT" ./mimpirun "$NUM_OF_WORKERS" "$@" 2> "$tmp_file" || true
 
 result=0
 
+if [ -z ${VALGRIND+x} ]; then
+    timeout "$TIMEOUT" ./mimpirun "$NUM_OF_WORKERS" "$@" 2> "$tmp_file" || true
+else
+
+    LC="${TIMEOUT: -1}"
+    if [ "$LC" = "s" ]; then
+    TIMEOUT=${TIMEOUT::-1}
+    fi
+    TIMEOUT=$(echo "$TIMEOUT * 20" | bc)
+
+    if [ $(echo "$TIMEOUT > 1000" | bc) = "1" ] ; then
+        echo "SKIPPING TEST::TOO LONG" >&2
```

Comment:
To się nie łączy z self testami tzn. wywalają się na tym outpucie (bo to nie jest ten, którego oczekiwały). Nie mam pomysłu jak to poprawić, więc chyba może być (ale może ktoś inny ma).

Reply:
Owszem, ale wywali się tylko dla tych testów które mają limit czasowy >50s, więc nie jest to jakiś bardzo duży problem.